### ISSUE TITLE
Support for constant byte literal

### ISSUE THREAD

**Issue by Wodann** [eqrion/cbindgen#546]
The code contains a [TODO](https://github.com/eqrion/cbindgen/blob/bb00d1c4a8e53c56880586987f0e1512fb252b9a/src/bindgen/ir/constant.rs#L220) for adding byte string support.

Is it on the roadmap and/or what would a preferred implementation look like?

My use case looks something like this:
```rust
pub const GET_INFO_FN_NAME: &'static u8 = b"get_info\0";
```

**Comment by emilio** [comment]
Yeah, this shouldn't be terrible to add I think, wanna give it a shot? I'm happy to help out.

Seems like generating a `const uint8_t[]` should be the most straight-forward implementation unless I'm missing something.

**Comment by Wodann** [comment]
Yes, I can start looking into it. It might take me a while, because I also have some other higher priority tasks for my open-source project.

W.r.t. design, in Rust one might write the following:
```rust
pub const NAME: &[u8] = b"test\0";
```

For which the equivalent, depending on chosen style and lang version, could be:
```cpp
constexpr char NAME[] = "test\0";
```

You might already notice that there is a type difference `u8` vs `c_char`. How do we want to handle that? Do we parse for the byte string literal pattern and assume it's intended to be a `c_char`?

**Comment by emilio** [comment]
Strings are implicitly null-terminated in C/C++, so you don't need the `"test\0"` there, just `"test"` would do in that case.

But I think you really need a `const uint8_t NAME[] = [x, y, z];` when generating code. Otherwise you can't represent stuff like `b"not null terminated"`, for example.

**Comment by lygstate** [comment]
Please use this to reproduce
https://github.com/rulibc/rulibc/tree/cbindgen

The branch cbindgen
